docs/en/guides/pipeline-integrations/development/sgtk-developer-generating-path-and-publish.md

```diff
@@ -0,0 +1,567 @@
+
+import sgtk
+import os
+
+# Get the engine instance that is currently running.
+current_engine = sgtk.platform.current_engine()
+
+# Grab the pre-created Sgtk instance from the current engine.
+tk = current_engine.sgtk
+
+# Get a context object from a Task, this Task must belong to a Shot for the future steps to work. 
+context = tk.context_from_entity("Task", 13155)
+
+# Create the required folders based upon the task
+tk.create_filesystem_structure("Task", context.task["id"])
+
+# Generating a Path
+# =================
+
+# Get a template instance by providing a name of a valid template in your config's templates.yml
+template = tk.templates["maya_shot_publish"]
+
+# Use the context to resolve as many of the template fields as possible.
+fields = context.as_template_fields(template)
+
+# Manually resolve the remaining fields that can't be figured out automatically from context.
+fields["name"] = "myscene"
+
+# Get an authenticated Shotgun API instance from the engine
+sg = current_engine.shotgun
+
+# Run a Shotgun API query to summarize the maximum version number on PublishedFiles that
+# are linked to the task and match the provided name.
+# Since PublishedFiles generated by the Publish app have the extension on the end of the name we need to add the
+# extension in our filter.
+r = sg.summarize(entity_type="PublishedFile",
+                 filters = [["task", "is", {"type":"Task", "id": context.task["id"]}],
+                            ["name","is", fields["name"] + ".ma"]],
+                 summary_fields=[{"field":"version_number", "type":"maximum"}])
+
+# Extract the version number and add 1 to it.
+# In scenarios where there are no files already this summary will return 0.
+fields["version"] = r["summaries"]["version_number"] + 1
+
+# Use the fields to resolve the template path into an absolute path.
+publish_path = template.apply_fields(fields)
+
+# Make sure we create any missing folders
+current_engine.ensure_folder_exists(os.path.dirname(publish_path))
+
+# Creating a file
+# ===============
+
+# This is the bit where you would add your own logic to copy or save a file using the path.
+# In the absence of any file saving in the example, we'll use the following to create an empty file on disk.
+sgtk.util.filesystem.touch_file(publish_path)
+
+# Publishing
+# ==========
+
+# So as to match publishes created by the publish app's, we are adding the extension to the end of the publish name.
+publish_name = fields["name"] + ".ma"
+version_number = fields["version"]
+
+# Now register the publish
+sgtk.util.register_publish(tk,
+                           context,
+                           publish_path,
+                           publish_name,
+                           version_number,
+                           published_file_type = "Maya Scene")
+```
+
+{% include info title="Tip" content="By this point, the code is getting a bit long, so a recommended next step would be to tidy it up a bit and break things into methods." %}
```

Comment:
So good! I feel like we've gone on a JOURNEY! Maybe some concluding lines? Additional resources, or point to the forum or something?

Reply:
good idea, I've added:

> ### Final Thoughts

This guide has hopefully left you with a foundational understanding of how to get started with the Toolkit API.
There is of course many other uses for the API, so we recommend reading through the [tk-core API](https://developer.shotgunsoftware.com/tk-core/index.html) for more information.

Also our [forums](https://community.shotgunsoftware.com/c/pipeline/6) are an excellent place to discuss API questions and get answers, and even leave feedback for us about the this guide.